baremaps-core/src/main/java/org/apache/baremaps/workflow/tasks/CreateIplocIndex.java

```diff
@@ -26,101 +26,93 @@
-      org.apache.baremaps.iploc.IpLoc ipLoc = new IpLoc(jdbcUrl, geocoder);
-      ipLoc.insertNicObjects(fetchNicObjectStream);
-      IpLocStats ipLocStats = ipLoc.getIplocStats();
-
-      logger.info(
-              """
-                      IpLoc stats
-                      -----------
-                      inetnumInsertedByAddress : {}
-                      inetnumInsertedByDescr : {}
-                      inetnumInsertedByCountry : {}
-                      inetnumInsertedByCountryCode : {}
-                      inetnumInsertedByGeoloc : {}
-                      inetnumNotInserted : {}""",
-              ipLocStats.getInsertedByAddressCount(), ipLocStats.getInsertedByDescrCount(),
-              ipLocStats.getInsertedByCountryCount(), ipLocStats.getInsertedByCountryCodeCount(),
-              ipLocStats.getInsertedByGeolocCount(), ipLocStats.getNotInsertedCount());
-
-      logger.info("IpLoc database created successfully");
-
-    logger.info("Finished creating the Geocoder index {}", targetIplocIndexPath);
-  }
-
-  @Override
-  public boolean equals(Object o) {
-    if (this == o) return true;
-    if (o == null || getClass() != o.getClass()) return false;
-    CreateIplocIndex that = (CreateIplocIndex) o;
-    return Objects.equals(geonamesIndexPath, that.geonamesIndexPath) && Arrays.equals(iplocNicPath, that.iplocNicPath) && Objects.equals(targetIplocIndexPath, that.targetIplocIndexPath);
-  }
-
-  @Override
-  public int hashCode() {
-    int result = Objects.hash(geonamesIndexPath, targetIplocIndexPath);
-    result = 31 * result + Arrays.hashCode(iplocNicPath);
-    return result;
-  }
-
-  @Override
-  public String toString() {
-    return "CreateIplocIndex{" +
-            "geonamesIndexPath='" + geonamesIndexPath + '\'' +
-            ", iplocNicPath=" + Arrays.toString(iplocNicPath) +
-            ", targetIplocIndexPath='" + targetIplocIndexPath + '\'' +
-            '}';
-  }
+
+        IpLocStats ipLocStats = ipLoc.getIplocStats();
+
+        logger.info(
+                """
+                        IpLoc stats
+                        -----------
+                        inetnumInsertedByAddress : {}
+                        inetnumInsertedByDescr : {}
+                        inetnumInsertedByCountry : {}
+                        inetnumInsertedByCountryCode : {}
+                        inetnumInsertedByGeoloc : {}
+                        inetnumNotInserted : {}""",
+                ipLocStats.getInsertedByAddressCount(), ipLocStats.getInsertedByDescrCount(),
+                ipLocStats.getInsertedByCountryCount(), ipLocStats.getInsertedByCountryCodeCount(),
+                ipLocStats.getInsertedByGeolocCount(), ipLocStats.getNotInsertedCount());
+
+        logger.info("IpLoc database created successfully");
+
+        logger.info("Finished creating the Geocoder index {}", targetIplocIndexPath);
+    }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
equals, hashCode, and toString are implemented by records. The default probably don't work becose of the nicPaths array. Maybe we can replace the array by a List<String>.

Reply:
I have changed the array to a list and it works fine now! Thank you!